packages/rum-core/src/domain/startCustomerDataTelemetry.spec.ts

```diff
@@ -0,0 +1,165 @@
+      })
+      .beforeBuild(({ globalContextManager, userContextManager, featureFlagContexts, configuration }) => {
+        batchFlushObservable = new Observable()
+        lifeCycle = new LifeCycle()
+        fakeContextBytesCount = 1
+        fakeContext = { foo: 'bar' }
+        spyOnContextManager(globalContextManager)
+        spyOnContextManager(userContextManager)
+        spyOnFeatureFlagContexts(featureFlagContexts)
+
+        telemetryEvents = []
+        const telemetry = startTelemetry(TelemetryService.RUM, configuration)
+        telemetry.observable.subscribe((telemetryEvent) => telemetryEvents.push(telemetryEvent))
+
+        startCustomerDataTelemetry(
+          configuration,
+          telemetry,
+          lifeCycle,
+          globalContextManager,
+          userContextManager,
+          featureFlagContexts,
+          batchFlushObservable
+        )
+      })
+  })
+
+  afterEach(() => {
+    setupBuilder.cleanup()
+    resetExperimentalFeatures()
+  })
+
+  it('should collect customer data telemetry', () => {
+    const { clock } = setupBuilder.build()
+
+    generateBatch({ eventNumber: 2, contextBytesCount: 2, batchBytesCount: 2 })
+    generateBatch({ eventNumber: 1, contextBytesCount: 1, batchBytesCount: 1 })
+    clock.tick(MEASURES_PERIOD_DURATION)
+
+    expect(telemetryEvents[0].telemetry).toEqual({
+      type: 'log',
+      status: 'debug',
+      message: 'Customer data measures',
+      batchCount: 2,
+      batchBytesCount: { min: 1, max: 2, sum: 3 },
+      batchMessagesCount: { min: 1, max: 2, sum: 3 },
+      globalContextBytes: { min: 1, max: 2, sum: 5 },
+      userContextBytes: { min: 1, max: 2, sum: 5 },
+      featureFlagBytes: { min: 1, max: 2, sum: 5 },
+    })
+  })
+
+  it('should not collect empty contexts telemetry', () => {
+    const { clock } = setupBuilder.build()
+
+    generateBatch({ eventNumber: 1, context: {} })
+
+    clock.tick(MEASURES_PERIOD_DURATION)
+
+    expect(telemetryEvents[0].telemetry.globalContextBytes).not.toBeDefined()
+    expect(telemetryEvents[0].telemetry.userContextBytes).not.toBeDefined()
+    expect(telemetryEvents[0].telemetry.featureFlagBytes).not.toBeDefined()
+  })
+
+  it('should not collect contexts telemetry of a unfinished batches', () => {
+    const { clock } = setupBuilder.build()
+
+    generateBatch({ eventNumber: 1, contextBytesCount: 1, batchBytesCount: 1 })
+    lifeCycle.notify(LifeCycleEventType.RUM_EVENT_COLLECTED, { type: RumEventType.VIEW } as RumEvent & Context)
+    clock.tick(MEASURES_PERIOD_DURATION)
+
+    expect(telemetryEvents[0].telemetry).toEqual(
+      jasmine.objectContaining({
+        batchCount: 1,
+        batchBytesCount: { min: 1, max: 1, sum: 1 },
+        batchMessagesCount: { min: 1, max: 1, sum: 1 },
+        globalContextBytes: { min: 1, max: 1, sum: 1 },
+        userContextBytes: { min: 1, max: 1, sum: 1 },
+        featureFlagBytes: { min: 1, max: 1, sum: 1 },
+      })
+    )
+  })
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏could we focus the test setup/expectation to make clearer what would fail if we collect contexts telemetry of unfinished batches?